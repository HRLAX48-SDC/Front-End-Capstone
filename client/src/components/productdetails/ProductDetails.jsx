import React, {useState, useEffect} from 'react';
import axios from 'axios';
import API_KEY from '../../config/config.js';
import Navbar from './Navbar.jsx';
import ProductInfo from './ProductInfo.jsx';

function ProductDetails(props) {

  const [info, setInfo] = useState({
    id: 0,
    campus: '',
    name: '',
    slogan: '',
    description: '',
    category: '',
    default_price: '',
    created_at: '',
    updated_at: '',
    features: []
  });

  const [style, setStyle] = useState({
    product_id: 0,
    results: []
  });

  const [cart, setCart] = useState([]);

  useEffect(() => {
    getInfo();
    getStyle();
    getCartInfo();
  }, []);

  const getInfo = () => {
    axios.get('https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax/products/42366', {
      headers: {
        'Authorization': `${API_KEY}`
      }
    })
    .then(response => {
      setInfo(response.data);
    })
    .catch(err => {
      console.error(err);
    })
  }

  const getStyle = () => {
    axios.get('https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax/products/42366/styles', {
      headers: {
        'Authorization': `${API_KEY}`
      }
    })
    .then(response => {
      setStyle(response.data);
    })
    .catch(err => {
      console.error(err);
    })
  }

  const getCartInfo = () => {
    axios.get('https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax/cart', {
      headers: {
        'Authorization': `${API_KEY}`
      }
    })
    .then(response => {
      setCart(response.data);
    })
    .catch(err => {
      console.error(err);
    })
  }

  return (
    <div>
      <Navbar />
      <ProductInfo info={info} style={style.results} />
    </div>
  )
}

export default ProductDetails;