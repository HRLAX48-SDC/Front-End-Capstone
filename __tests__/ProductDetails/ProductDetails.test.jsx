import React from 'react';
import Enzyme, { configure, mount, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import ProductDetails from '../../client/src/components/productdetails/ProductDetails.jsx';

const relatedItems = [];

test('Renders no item', () => {
  const wrapper = mount(<ProductDetails product={{}} />);
  expect(wrapper).toHaveProp({product: {}});
})


describe('<ProductDetails />', () => {
  const setState = jest.fn();
  const useStateMock = () => [initState, setState];

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Is a test where we want to mock useState', () => {
        jest.spyOn(React, 'useState').mockImplementation(useStateMock);
        const wrapper = shallow(<ProductDetails product={{id: 42366,
          campus: "hr-lax",
          name: "Camo Onesie",
          slogan: "Blend in to your crowd",
          updated_at: "2021-08-13T14:39:39.968Z",
          category: "Jackets",
          created_at: "2021-08-13T14:39:39.968Z",
          default_price: "140.00",
          description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
         }} ratings={{}} />);
        expect(setState).toHaveBeenCalledTimes(0);
  });
});

