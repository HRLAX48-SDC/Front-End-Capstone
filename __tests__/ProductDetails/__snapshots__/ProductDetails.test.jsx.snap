// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Product Details component correctly 1`] = `
<div>
  <nav
    className="navbar navbar-light bg-light p_navbar"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
      >
        Logo
      </a>
      <form
        className="d-flex"
      >
        <input
          aria-label="Search"
          className="form-control me-2"
          placeholder="Search"
          type="search"
        />
        <button
          className="btn btn-outline-gray"
          type="submit"
        >
          <i
            className="fas fa-search"
          />
        </button>
      </form>
    </div>
  </nav>
  <div
    className="p_announcement"
  >
    SITE-WIDE ANNOUNCEMENT MESSAGE! -- SALE / DISCOUNT 
    <span
      className="p_offer"
    >
      OFFER
    </span>
     -- 
    <span
      className="p_highlight"
    >
      NEW PRODUCT HIGHLIGHT
    </span>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <div
        className="product-grid"
      >
        <div
          className="row justify-content-around"
        >
          <div
            className="col-md-2"
          >
            <button
              className="p_up"
              onClick={[Function]}
            >
              <i
                className="fas fa-chevron-up"
              />
            </button>
            <button
              className="p_down"
              onClick={[Function]}
            >
              <i
                className="fas fa-chevron-down"
              />
            </button>
          </div>
          <div
            className="col-md-9"
          >
            <div
              className="spinner-border"
              role="status"
            >
              <span
                className="visually-hidden"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-content col-md-5"
    >
      <div />
      <h4>
        Jackets
      </h4>
      <h2
        className="p_title"
      >
        <a
          href="#"
        >
          Camo Onesie
        </a>
      </h2>
      <div
        className="p_price"
      />
      <div
        className="p_style"
      >
        <b>
          STYLE &gt; 
        </b>
        <span />
        <div
          className="p_thumbnail_container"
        >
          <div
            className="spinner-border"
            role="status"
          >
            <span
              className="visually-hidden"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
      <div>
        <div
          className="p_flex_container"
        >
          <div
            className="p_flex_child_size"
          >
            <select
              className="p_select"
              onChange={[Function]}
            >
              <option>
                SELECT SIZE
              </option>
              <option>
                OUT OF STOCK
              </option>
            </select>
          </div>
          <div
            className="p_flex_child_quantity"
          >
            <select
              className="p_select"
            >
              <option>
                -
              </option>
            </select>
          </div>
        </div>
        <button
          className="btn btn-default btn-md p_fav btn btn-light"
          disabled={false}
          id="p_like_button"
          onClick={[Function]}
          type="button"
        >
          <i
            className="far fa-star"
          />
        </button>
      </div>
    </div>
    <div
      className="p_description"
    >
      <h6>
        DESCRIPTION:
      </h6>
      <p>
        The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.
      </p>
      <div>
        <span>
          Share: 
        </span>
        <a
          className="p_share_icon"
          href="https://www.facebook.com/"
        >
          <i
            className="fab fa-facebook-square fa-lg"
          />
        </a>
        <a
          className="p_share_icon"
          href="https://twitter.com/home"
        >
          <i
            className="fab fa-twitter-square fa-lg"
          />
        </a>
        <a
          className="p_share_icon"
          href="https://www.pinterest.com/"
        >
          <i
            className="fab fa-pinterest-square fa-lg"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
